spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindingTargetFactory.java

```diff
@@ -19,26 +19,35 @@
 import org.springframework.messaging.SubscribableChannel;
 
 /**
- * Defines methods to create/configure the {@link org.springframework.messaging.MessageChannel}s defined
- * in {@link org.springframework.cloud.stream.annotation.EnableBinding}.
+ * Defines methods to create/configure the bound elements defined by
+ * {@link org.springframework.cloud.stream.annotation.EnableBinding}.
+ *
+ * @author Marius Bogoevici
  * @author Ilayaperumal Gopinathan
  */
-public interface BindableChannelFactory {
+public interface BindingTargetFactory {
 
 	/**
-	 * Create an input {@link SubscribableChannel} that will be bound via
-	 * the message channel {@link org.springframework.cloud.stream.binder.Binder}.
-	 * @param name name of the message channel
-	 * @return subscribable message channel
+	 * Checks whether a specific type bound element can be created by this factory.
+	 * @param clazz the type of the bound element
+	 * @return true if the bound element can be created
+	 */
+	boolean canCreate(Class<?> clazz);
+
+	/**
+	 * Create an input bindable element that will be bound via a corresponding binder
+	 * {@link org.springframework.cloud.stream.binder.Binder}.
+	 * @param name name of the bindable element
+	 * @return bindable element
 	 */
-	SubscribableChannel createInputChannel(String name);
+	Object createInput(String name);
 
 	/**
-	 * Create an output {@link SubscribableChannel} that will be bound via
-	 * the message channel {@link org.springframework.cloud.stream.binder.Binder}.
+	 * Create an output {@link SubscribableChannel} that will be bound via the message
+	 * channel {@link org.springframework.cloud.stream.binder.Binder}.
```

Comment:
should not mention "message channel" here (looks like the corresponding input method's comment was genericized, but this one was not)